openslides_backend/action/actions/user/save_saml_account.py

```diff
@@ -465,10 +459,26 @@ def get_field_data(
                             )
                         )
                     elif saml_meeting_user_field == "vote_weight":
-                        # Result is string and has 6 digits.
-                        if isinstance(value, str):
-                            value = float(value)
-                        result = f"{value:f}"
+                        # Result must be string and have 6 digits after dot.
+                        if isinstance(value, int):
+                            value = f"{value:f}"
+                        elif isinstance(value, str) and re.compile(
+                            r"^[1-9]\d*\.?\d{0,6}$"
```

Comment:
I do not believe this will accept `0.***...` values (or negative values for that matter, but those don't make sense for the vote_weight anyway). I get that you need the fraction part to be variable length and therefore cannot use the `DECIMAL_REGEX` outright, but maybe you could replicate its format more closely